creators/stops_creator.py

```diff
@@ -20,18 +21,41 @@ def add_stops_to_schedule(self, schedule, data):
         stops = data.get_stops()
 
         # add all stops to GTFS
-        for stop in stops.values():
+        for elem in stops.values():
+            if type(elem) is StopArea:
+                if len(elem.stop_members) > 1:
+                    parent_station = self.add_stop(schedule, elem, None, True)
+                    for stop in elem.stop_members.values():
+                        self.add_stop(schedule, stop, parent_station)
+                else:
+                    stop = elem.stop_members.values()[0]
+                    self.add_stop(schedule, stop)
+            else:
+                self.add_stop(schedule, elem)
+
+        # Add loose stop objects to route objects
+        self.add_stops_to_routes(data)
 
+    def add_stop(this, schedule, stop, parent_station=None, is_station=False):
             # Add stop to GTFS object
-            schedule.AddStop(
+            stop_obj = schedule.AddStop(
                 lat=float(stop.lat),
                 lng=float(stop.lon),
                 name=stop.name,
                 stop_id=str(stop.id)
```

Comment:
maybe add an prefix or suffix to the id ?
You could find some edge cases where a relation has the same id as a node : if this happens, you may have issues as the stop_id must be unique.

Reply:
Good point! Only problem is that it would break (my) existing stop IDs. How about only adding a prefix for stop_areas then?

Reply:
:+1:

Reply:
Ok. I'll add the prefix "SA" to the stop_area only.

Reply:
Indeed its going to break Fenix code as it happened with incofer's. Maybe could be fixed as I did in the incofer trips creator.

Reply:
@grote I ran fenix code and it works. I think this is because all `stop_area` relations in [floriapa bbox](http://overpass-turbo.eu/s/s8Y) have only one platform element (all relations have 1 stop and 1 platform as members.). With this data the `stop_creator` will not add stops as *parent stations* (based stop areas). But just in case, I committed a suggested modification to the `trips_creator_fenix` in order to manage eventual *parent stations*.

Reply:
Thanks for checking the fenix code @jamescr!

> With this data the stop_creator will not add stops as parent stations (based stop areas).

What do you mean by that? The `stop_creator` will just ignore those stop area relations?

In the code, it looks like `len(elem.stop_members) > 1` will still be true in that case.

> But just in case, I committed a suggested modification to the trips_creator_fenix in order to manage eventual parent stations.

This change seems to ignore stop areas completely, right?

Incomplete stop area relations might be common in other data sets as well. Shouldn't the default stop creator handle them well then?

Reply:
>> With this data the stop_creator will not add stops as parent stations (based stop areas).

> What do you mean by that? The stop_creator will just ignore those stop area relations?
> In the code, it looks like len(elem.stop_members) > 1 will still be true in that case.

The overpass query retrieves stops tagged as "platform" and its corresponding "stop_area" relation (with its members). Happens that floriapa "stop_area"' relations has two members, one tagged as "platform" and another tagged as "stop_position". When the `StopArea` object is created, member tagged as `stop_position` is ignored because is not a *valid stop candidate*. Given this, in this case the total `StopArea.stop_members` is 1. 

In this cases, adding the `stop_area` as a "gtfs stop station" and a `stop`as a "gtfs stop" makes no sense, that is why the `stop_area` is ignored and only the  `stop` is added to the stops.txt. If the `stop_area` is updated with more members tagged as `platform`  the creator will include the stop_area and each stop (as is going on incofer case). 


>> But just in case, I committed a suggested modification to the trips_creator_fenix in order to manage eventual parent stations.

> This change seems to ignore stop areas completely, right?

Yes. Was just preventing from a break that will never occur. (I'll explain at the end of the comment).

> Incomplete stop area relations might be common in other data sets as well. Shouldn't the default stop creator handle them well then?

Yes, the stop creator would not break if data is incomplete.

---

The type of the list of `stops` used in both trip_creators is `Stop` never a `StopArea`. That is clear when you look at the `stops_creator._get_stop` method, always will return a `Stop`.

Unfortunately for me I tested incofer trip creator with outdated data, when testing fenix with fresh data and back to incofer also with fresh data, everything was ok with the actual trip creators.

Reply:
Thanks for the explanation! :)

And now let's get this finally merged!